### PR TITLE
Changed docu for HTTP Request - not using $container->getByType(), bu…

### PR DESCRIPTION
**Pull request by paveljanda**
…t our own service class and its contructor

### CODE REVIEW

**Comment by dg** [comment]
Because this applies to all services (session, cache), better is the link to https://doc.nette.org/cs/2.4/di-usage, isn't?

**Comment by paveljanda** [comment]
@dg Oh I see.

Yes it is. But I am not really happy to see the `Container::getByType()` in that place. Event if there is a link to `di-usage` (As you can see, I didn't really noticed it. I was not trying to though).

I would be glad to see there a usage in some class registered in DIC. Without the comment I added and the snippet with a Presenter. Hm?

**Comment by dg** [comment]
I totally agree that getByType should not be there.

**Comment by paveljanda** [comment]
I removed my mentions about DIC, but left there a class constructor.
I think it is important to leave here a note about Presenter and request/response objects.

**Comment by dg** [comment]
ad presenter: 
I think that there is no need for example how to use `getHttpRequest()` in presenter. It is enough to mention the method.

ad contructor:
It is strange to describe on every HTTP, cache, session, etc. page how DIC works, how the configuration file works, how pass dependency etc. One paragraph with link to one common page is imho better.

File: cs/http-request-response.texy [hunk 1]
```diff
@@ -13,7 +13,48 @@ Nette Framework pročistí data odeslané uživatelem od kontrolních a neplatn
...
+
+	public function __construct(Request $httpRequest)
+	{
+		$this->httpRequest = $httpRequest;
+	}
+
+}
+\--
+
+Presenter má HTTP request k dispozici v předkovi. Vyžádat si ho můžeme metodou `Nette\Application\UI\Presneter::getHttpRequest()`:
```
Comment (f3l1x) [inline cs/http-request-response.texy]: Typo `Presneter` -> `Presenter`.

File: en/http-request-response.texy [hunk 2]
```diff
@@ -14,7 +14,48 @@ Nette Framework cleans out data sent by user from control and invalid characters
...
+
+	public function __construct(Request $httpRequest)
+	{
+		$this->httpRequest = $httpRequest;
+	}
+
+}
+\--
+
+Presenter has the HTTP request available in parent class. It can ask for it using method `Nette\Application\UI\Presneter::getHttpRequest()`:
```
Comment (f3l1x) [inline en/http-request-response.texy]: Typo

File: cs/http-request-response.texy [hunk 3]
```diff
@@ -13,7 +13,48 @@ Nette Framework pročistí data odeslané uživatelem od kontrolních a neplatn
...
+
+/--php
+<?php
```
Comment (f3l1x) [inline cs/http-request-response.texy]: I think it's not needed to show presenter with `getHttpRequest`. It's mention right above. WDYT @paveljanda ?

**Comment by dg** [comment]
Simplified and merged  adf7a22a4d7013d554a848a7851d5ed438d1ec27

**Comment by f3l1x** [comment]
Great. Thanks @dg

**Comment by paveljanda** [comment]
Thanks! Sorry for no responses from me, I was away from the computer for a few days

**Pull request closed without merging**